channeldb/channel.go

```diff
@@ -3252,6 +3301,441 @@ const (
+// DeserializeChainActions reads out the chainAction map from the passed
+// reader.
+func DeserializeChainActions(r io.Reader) (map[string][]HTLC,
+	error) {
+
+	chainAction := make(map[string][]HTLC)
+
+	tlvStream, err := tlv.NewStream(
+		tlv.MakeDynamicRecord(
+			chainActionType, &chainAction,
+			HtlcActionRecordSize(&chainAction),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+	)
+
+	if err != nil {
+		return chainAction, err
+	}
+
+	var bodyLen int64
+	err = binary.Read(r, byteOrder, &bodyLen)
+	if err != nil {
+		return chainAction, err
+	}
+
+	lr := io.LimitReader(r, bodyLen)
+	if err = tlvStream.Decode(lr); err != nil {
+		return chainAction, err
+	}
+
+	return chainAction, nil
+}
+
+// serializeLocalForceCloseInsights writes out the passed
+// localForceCloseInsights to the writer.
+func serializeLocalForceCloseInsights(w io.Writer,
+	lc *LocalForceCloseInsights) error {
+
+	localForceCloseReasonByte := []byte(lc.Initiator)
+
+	tlvStream, err := tlv.NewStream(
+		tlv.MakePrimitiveRecord(localForceCloseInitiatorType,
+			&localForceCloseReasonByte),
+		tlv.MakeDynamicRecord(
+			chainActionType, &lc.ChainActions,
+			HtlcActionRecordSize(&lc.ChainActions),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+		tlv.MakePrimitiveRecord(broadCastHeightType,
+			&lc.BroadcastHeight),
+	)
+
+	if err != nil {
+		return err
+	}
+	var b bytes.Buffer
+	err = tlvStream.Encode(&b)
+	if err != nil {
+		return err
+	}
+
+	err = binary.Write(w, byteOrder, uint64(b.Len()))
+	if err != nil {
+		return err
+	}
+
+	if _, err = w.Write(b.Bytes()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deSerializeLocalForceCloseInsights reads out the LocalForceCloseInsights from
+// the passed reader.
+func deSerializeLocalForceCloseInsights(r io.Reader) (*LocalForceCloseInsights,
+	error) {
+
+	var (
+		initiator []byte
+		height    uint32
+	)
+
+	chainAction := make(map[string][]HTLC)
+	tlvStream, err := tlv.NewStream(
+		tlv.MakePrimitiveRecord(localForceCloseInitiatorType,
+			&initiator),
+		tlv.MakeDynamicRecord(
+			chainActionType, &chainAction,
+			HtlcActionRecordSize(&chainAction),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+		tlv.MakePrimitiveRecord(broadCastHeightType,
+			&height),
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var bodyLen int64
+	err = binary.Read(r, byteOrder, &bodyLen)
+	if err != nil {
+		return nil, err
+	}
+
+	lr := io.LimitReader(r, bodyLen)
+	if err = tlvStream.Decode(lr); err != nil {
+		return nil, err
+	}
+
+	fcInsights := LocalForceCloseInsights{
+		Initiator:       LocalForceCloseInitiator(initiator),
+		ChainActions:    chainAction,
+		BroadcastHeight: height,
+	}
+
+	return &fcInsights, nil
+}
+
+// attachLocalForceCloseInsights fetches the local force close insights
+// previously stored when commitment was broadcast.
+func attachLocalForceCloseInsights(chanBucket kvdb.RBucket,
+	channel *OpenChannel) error {
+
+	insightBytes := chanBucket.Get(localFCInsightsKey)
+	if insightBytes == nil {
```

Comment:
I think even if it is a remote FC the channel would not have the `ChanStatusCommitBroadcasted` bit in the status. While fetching historical buckets we also use this function and then the  `ChanStatusCommitBroadcasted` bit would be in the status but we do not store the force close insight in this bucket so it would be nil. I would update the comment in this line to reflect that.